lib/intl.ts

```diff
@@ -341,17 +453,17 @@ function instantAmend(optionsParam) {
   return options;
 }
 
-function hasDateOptions(options) {
+function hasDateOptions(options: Parameters<TypesWithToLocaleString['toLocaleString']>[1]) {
   return 'year' in options || 'month' in options || 'day' in options || 'weekday' in options || 'dateStyle' in options;
 }
 
-function hasTimeOptions(options) {
+function hasTimeOptions(options: Parameters<TypesWithToLocaleString['toLocaleString']>[1]) {
   return (
     'hour' in options || 'minute' in options || 'second' in options || 'timeStyle' in options || 'dayPeriod' in options
   );
 }
 
-function isTemporalObject(obj) {
+function isTemporalObject(obj: unknown) {
```

Comment:
Nit: Is the returned type being inferred correctly here? Maybe it would be better to explicitly type is as `value is AnyTemporalType` (or whatever the right combination actually is)?